hyperspy/events.py

```diff
@@ -54,16 +103,51 @@ def suppress(self):
             obj.val_b = b
         obj.events.values_changed.trigger()
         """
-        return EventSuppressionContext(self)
+        return EventsSuppressionContext(self)
 
 
 class Event(object):
 
     def __init__(self):
         self._connected = {}
-        self.suppress = False
+        self._suppress = False
+
+    def suppress(self, function=None):
```

Comment:
I originally designed it split into two, but merged them. I've split them back up again.